ChromProcess/Utils/peak_finding/pick_peaks.py

```diff
@@ -1,6 +1,96 @@
 import numpy as np
 from ChromProcess.Utils.signal_processing import signal_processing as sig
 
+
+def find_peak_boundaries(signal, peaks_indices, peak_window=1):
```

Comment:
I like this refactor, let's keep it.

Reply:
Interesting point about it stopping at the ends of the time array passed to it. I've never had this issue, since I always gave peaks enough padding in each region. Maybe it's worth checking for this in the Processing module.

Reply:
It's a problem that results from the use of the peak_window. And possibly a side effect of how strictly you control peak thresholds (a problem I had in my last dataset), `signal[cursor:cursor+peak_window]` will give an error if `cursor+peak_window > len(signal)` so it's a bit sloppy to leave it in. You solved this with `int(cursor) >= len(diff)`
Putting it in the while loop prevents the extra if statement, disadvantage is that I had to make it a double while loop to not make it sloppy in case the `peak_window` is very long (and alternative would be to find the location of the minimum once the edge is reached but this would be slightly more complex)

Reply:
OK `find_peak_boundaries` needs a bit of work, then. Does your updated code work with the boundary finding code in `pick_peaks`? If so, let's refactor that section as a boundary search function..

Reply:
I rewrote the function to now be based on finding the minimum index, this reduces the second while loop and hopefully improves readability. I think this function should speed up the process with a larger peak window, but it might be a bit slower with a peak window of 1.
I adapted  `find_peaks_in_region` to use the `peak_indices_to_time` and `find_peaks` to use the new `find_peak_boundaries` I get slightly ofset (one index value) I checked in the raw data but the new value appears to actually be the correct one. I wasn't sure what was causing this before but you might want to verify.